openassessment/xblock/static/js/src/lms/oa_response.js

```diff
@@ -801,25 +772,23 @@ export class ResponseView {
 
      */
    removeUploadedFile(filenum) {
-     const view = this;
-     return view.confirmRemoveUploadedFile(filenum).done(() => view.server.removeUploadedFile(filenum).done(() => {
-       const sel = $('.step--response', view.element);
-       const block = sel.find(`.submission__answer__file__block__${filenum}`);
-       block.html('');
-       block.prop('deleted', true);
-       view.checkSubmissionAbility();
-     }).fail((errMsg) => {
-       view.baseView.toggleActionError('delete', errMsg);
-     }));
+     if (this.confirmRemoveUploadedFile(filenum)) {
```

Comment:
This formerly relied on another `$.Deferred()`-wrapped `window.confirm()` and having both paradigms in the tests was gross, so I updated this case as well.